src/librustc_trans/back/link.rs

```diff
@@ -613,28 +616,32 @@ fn link_rlib<'a>(sess: &'a Session,
                 let bc_deflated_filename = obj_filename.with_extension(
                     &format!("{}.bytecode.deflate", i));
 
-                let bc_data = match fs::File::open(&bc_filename).read_to_end() {
-                    Ok(buffer) => buffer,
+                let mut bc_data = Vec::new();
+                match fs::File::open(&bc_filename).and_then(|mut f| {
```

Comment:
Ah yes I forgot about that, but with the new signature of `read_to_end` I suspect that may be more ergonomic.